### ISSUE TITLE
VictoryBoxPlot does not show label tooltips

### ISSUE THREAD

**Issue by harrisonlo** [FormidableLabs/victory#2621]
**Describe the bug**
The `labelComponent` prop does not seem to have an effect on VictoryBoxPlot, unlike the pattern in other charts such as VictoryPie or VictoryBar. Passing VictoryTooltip in would not show a tooltip on hover.

**Victory version**
36.6.10

**Code Sandbox link**
https://codesandbox.io/s/victory-boxplot-tooltip-issue-33tmtt

**To Reproduce**
1. Open sandbox
2. Hover around VictoryBoxPlot
3. Hover around VictoryBar to see tooltips

**Expected behavior**
VictoryBoxPlot with a VictoryTooltip component in the `labelComponent` prop would show a tooltip on hover

**Comment by troynguyen8** [comment]
I'm having a lot of difficulty with this too. Does anyone have a workaround, or is there any update?

**Comment by carbonrobot** [comment]
The `VictoryBoxPlot` does not have a `labelComponent` property, but instead has specific labels depending on the type of data you want to display. These label properties do not yet support the Tooltip.

**Comment by troynguyen8** [comment]
Hey @Burnett2k !

> it might be preferable to add support for tooltips in each of those labelComponents, rather than adding a new generic labelComponent. Would that work for you?

This would work well for my use case! I'm essentially hoping for a click on anywhere in the boxplot to show the stats in detail view. Within the tooltip could be something like
```
Full qualitative X axis name
min: 0
q1: 10
median: 15
q3: 20
max: 25
```

In an ideal world this tooltip would always point toward the median of the boxplot, but it would not be a deal breaker if I had to pass tooltip to each of the 5 stats and they'd point to the whisker/box lines all with the same content

**Comment by chriszrc** [comment]
I was really expecting that at a minimum, supplying supplying the tooltip to the specific labels:

`maxLabelComponent={<VictoryTooltip />}`

would work, but it doesn't 😥. My boxplots look a bit busy with all the labels, I think my users expect to hover for values, just like stacked bar charts or anything else:

<img width="649" alt="Screenshot 2024-05-28 at 4 28 33 PM" src="https://github.com/FormidableLabs/victory/assets/25695785/b34e7eff-da5b-4c60-bf7a-634c2886b62c">

Is there any way to work around this at the moment?

**Comment by carbonrobot** [comment]
@chriszrc Agreed, I think I was surprised its not supported as well! I have it up at the top of our features list, but currently no one available to tackle it. I don't have any easy workarounds atm, unfortunately.

**Comment by nlkluth** [comment]
This issue is still ongoing, but I have added a small workaround doc that will hopefully unblock some of you. The `events` property can be used to enable tooltips. See the doc here: https://github.com/FormidableLabs/victory/blob/main/website/docs/charts/box-plot.mdx#box-plot---tooltips.

**Comment by carbonrobot** [comment]
Since we have a current workaround, placing this issue on hold until the new events system is built which makes solving this trivial. No timeline on that build at this time.
